### PR TITLE
Feature/ebay

### PR DESCRIPTION
**Pull request by anton-karlovskiy**
eBay project is ready for review.

### CODE REVIEW

**Comment by addyosmani** [comment]
Thanks for the PR. Responding to your comment from the other thread, I believe what we have here is close enough to the original eBay for a demo. We don't need to copy the look/feel completely :)

**Comment by anton-karlovskiy** [comment]
@addyosmani 
When you review this PR, please give some feedback at the project README.

**Comment by anton-karlovskiy** [comment]
> Thanks for the PR. Responding to your comment from the other thread, I believe what we have here is close enough to the original eBay for a demo. We don't need to copy the look/feel completely :)

Yes, sounds good.
https://adaptive-loading.web.app/react-ebay-network-aware-code-splitting/
From this we can check the demo on firebase.

**Comment by anton-karlovskiy** [comment]
@addyosmani 
For network aware code splitting like this, we don't use React hooks but the lazy loading pattern with `navigator.connection.effectiveType` directly.

```
const Product = lazy(() => {
  return new Promise(resolve => {
    navigator.connection ? resolve(navigator.connection.effectiveType) : resolve(null);
  }).then(
    effectiveType => {
      console.log('effectiveType => ', effectiveType);
      switch (effectiveType) {
        case '4g':
          return import(/* webpackChunkName: "magnific-product" */ './MagnificProduct');
        case '3g':
        case '2g':
        case 'slow-2g':
          return import(/* webpackChunkName: "simple-product" */ './SimpleProduct');
        default:
          return import(/* webpackChunkName: "magnific-product" */ './MagnificProduct');
      }
    }
  );
});
```

**Comment by addyosmani** [comment]
> For network aware code splitting like this, we don't use React hooks but the lazy loading pattern with navigator.connection.effectiveType directly.

I think that's fine!

File: react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/MagnificProduct/index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+
+import React, { useState } from 'react';
+import ReactImageMagnify from 'react-image-magnify';
+import Modal from 'react-responsive-modal';
+
+import SimpleProduct from '../SimpleProduct';
+import './magnific-product.css';
+
+const MagnificProduct = ({ smallImageSrc, largeImageSrc, altMessage, externalRender, setIsHoverOnProduct }) => {
```
Comment (addyosmani) [inline react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/MagnificProduct/index.js]: `MagnificProduct` => `MagnifyProduct`

File: react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/MagnificProduct/index.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (addyosmani) [inline react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/MagnificProduct/index.js]: Can we change the component/module name to `MagnifyProduct`?

File: react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/MagnificProduct/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import React, { useState } from 'react';
+import ReactImageMagnify from 'react-image-magnify';
+import Modal from 'react-responsive-modal';
+
+import SimpleProduct from '../SimpleProduct';
+import './magnific-product.css';
```
Comment (addyosmani) [inline react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/MagnificProduct/index.js]: `magnific-` => `magnify-`

File: react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/ProductSummary/LeftSummary/index.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import PropertyField from '../../PropertyField';
+import {
+  ConditionPropertyValue,
+  USSizePropertyValue,
+  QuantityPropertyValue,
+  ShippingPropertyValue,
+  DeliveryPropertyValue,
+  PaymentsPropertyValue,
+  ReturnsPropertyValue
+} from './assemblies';
```
Comment (addyosmani) [inline react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/ProductSummary/LeftSummary/index.js]: Could we give assemblies a more descriptive name?

File: react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/MagnificProduct/index.js (hunk 2 above)
Comment (anton-karlovskiy) [reply react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/MagnificProduct/index.js]: Yes, of course.
I think naming is important to improve the readability.

File: react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/ProductSummary/RightSummary/ShopWithConfidence/shop-with-confidence.css [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+.dollar-mark {
+  margin-right: 14px;
+}
+
+.dollar-mark > span {
+  width: 21px;
+  height: 25px;
+  background-size: 44px;
+  display: inline-block;
+  background-image: url(https://ir.ebaystatic.com/rs/v/vswfg1if4iy2zhp0e3dxhytivik.png);
```
Comment (addyosmani) [inline react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/ProductSummary/RightSummary/ShopWithConfidence/shop-with-confidence.css]: Let's self-host this image locally

File: react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/MagnificProduct/index.js (hunk 1 above)
Comment (anton-karlovskiy) [reply react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/MagnificProduct/index.js]: Sure thing.

File: react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/MagnificProduct/index.js (hunk 3 above)
Comment (anton-karlovskiy) [reply react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/MagnificProduct/index.js]: Sure

File: react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/index.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+const Product = lazy(() => {
+  return new Promise(resolve => {
+    navigator.connection ? resolve(navigator.connection.effectiveType) : resolve(null);
+  }).then(
+    effectiveType => {
+      console.log('effectiveType => ', effectiveType);
+      switch (effectiveType) {
+        case '4g':
+          return import(/* webpackChunkName: "magnific-product" */ './MagnificProduct');
```
Comment (addyosmani) [inline react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/index.js]: Similar to the feedback earlier, let's update these values: ` return import(/* webpackChunkName: "magnify-product" */ './MagnifyProduct');` etc

File: react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/ProductSummary/LeftSummary/index.js (hunk 4 above)
Comment (anton-karlovskiy) [reply react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/ProductSummary/LeftSummary/index.js]: Yes, it sounds less descriptive as they are just for the purpose of demo elements.
Do you have some thing in your mind?

I'd suggest `SubComponents` or something.

File: react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/ProductSummary/RightSummary/ShopWithConfidence/shop-with-confidence.css (hunk 5 above)
Comment (anton-karlovskiy) [reply react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/ProductSummary/RightSummary/ShopWithConfidence/shop-with-confidence.css]: OK, no problem.

File: react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/index.js (hunk 6 above)
Comment (anton-karlovskiy) [reply react-ebay-network-aware-code-splitting/src/components/CenterPanelInternal/index.js]: Sure, I'll update all relevant parts not just sticking to the comments. :)

**Comment by anton-karlovskiy** [comment]
> Overall LGTM with some very minor comments 👍

Thank you for reviewing.

**Comment by anton-karlovskiy** [comment]
@addyosmani 
All comments are handled. Please review again and let's get this merged soon for the upcoming PRs.

**Pull request merged**